src/ucp/rndv/rndv_rtr.c

```diff
@@ -155,13 +155,14 @@ static size_t ucp_proto_rndv_rtr_pack_with_rkey(void *dest, void *arg)
     rpriv = req->send.proto_config->priv;
 
     ucs_assert(dt_iter->dt_class == UCP_DATATYPE_CONTIG);
-    ucs_assert(rpriv->super.md_map == dt_iter->type.contig.reg.md_map);
+    ucs_assert(ucs_test_all_flags(dt_iter->type.contig.memh->md_map,
+                                  rpriv->super.md_map));
 
     ucp_proto_rndv_rtr_hdr_pack(req, rtr, dt_iter->type.contig.buffer);
 
     rkey_size = ucp_proto_request_pack_rkey(req, rpriv->super.sys_dev_map,
                                             rpriv->super.sys_dev_distance,
-                                            rtr + 1);
+                                            rtr + 1, rpriv->super.md_map);
```

Comment:
make md_map the 2nd argument of ucp_proto_request_pack_rkey (after req) since it's 2nd most important